mono/tests/monitor-abort.cs

```diff
@@ -0,0 +1,41 @@
+using System;
+using System.Threading;
+
+public class Program {
+	const int num_threads = 10;
+	static int aborted = 0;
+
+	public static void ThreadFunc (object lock_obj)
+	{
+		try {
+			Monitor.Enter (lock_obj);
+			Monitor.Exit (lock_obj);
+		} catch (ThreadAbortException) {
+			Interlocked.Increment (ref aborted);
+		}
+	}
+
+	public static int Main (string[] args)
+	{
+		Thread[] tarray = new Thread [num_threads];
+
+		for (int i = 0; i < num_threads; i++) {
+			object lock_obj = new object ();
+			Monitor.Enter (lock_obj);
+			tarray [i] = new Thread (new ParameterizedThreadStart (ThreadFunc));
+			tarray [i].Start (lock_obj);
+		}
+
+		for (int i = 0; i < num_threads; i++)
+			tarray [i].Abort ();
```

Comment:
I just tried this and there's a race condition here: e.g. on my machine it prints "Aborted 0", because presumably the threads are aborted before they're started and had a chance to enter the lock.

Adding a `Thread.Sleep (1000);` before this loop fixes it for me and now it prints "Aborted 10".